src/py/flwr/server/superlink/state/sqlite_state.py

```diff
@@ -451,6 +452,36 @@ def get_task_res(self, task_ids: set[UUID], limit: Optional[int]) -> list[TaskRe
         if limit is not None and limit < 1:
             raise AssertionError("`limit` must be >= 1")
 
+        # Check if corresponding TaskIns exists and is not expired
+        task_ids_placeholders = ",".join([f":id_{i}" for i in range(len(task_ids))])
+        query = f"""
+            SELECT *
+            FROM task_ins
+            WHERE task_id IN ({task_ids_placeholders})
+            AND (created_at + ttl) > CAST(strftime('%s', 'now') AS REAL)
+        """
+        query += ";"
+
+        task_ins_data = {}
+        for index, task_id in enumerate(task_ids):
+            task_ins_data[f"id_{index}"] = str(task_id)
+
+        task_ins_rows = self.query(query, task_ins_data)
+
+        if not task_ins_rows:
+            log(ERROR, "TaskIns does not exist.")
+            return []
```

Comment:
Same as above

Reply:
I think we need to return empty here is this indicates that `task_ins_rows` is empty so to `task_res` to return.

Reply:
the two state implementations should not behave inconsistently. let's add a test that covers this particular scenarios (some ids are valid, while others are either expired or unknown) and then align the two implementations.